src/main/java/edu/cornell/kfs/fp/batch/service/impl/AccountingXmlDocumentDownloadAttachmentServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+package edu.cornell.kfs.fp.batch.service.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLConnection;
+import java.util.Collection;
+
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.IOUtils;
+import org.kuali.kfs.krad.bo.Attachment;
+import org.kuali.kfs.krad.document.Document;
+import org.kuali.kfs.krad.service.AttachmentService;
+import org.springframework.beans.factory.DisposableBean;
+
+import edu.cornell.kfs.fp.batch.service.AccountingXmlDocumentDownloadAttachmentService;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentBackupLink;
+import edu.cornell.kfs.sys.businessobject.WebServiceCredential;
+import edu.cornell.kfs.sys.service.WebServiceCredentialService;
+import edu.cornell.kfs.sys.service.impl.DisposableClientServiceImplBase;
+
+public class AccountingXmlDocumentDownloadAttachmentServiceImpl extends DisposableClientServiceImplBase implements AccountingXmlDocumentDownloadAttachmentService, DisposableBean {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(AccountingXmlDocumentDownloadAttachmentServiceImpl.class);
+
+    protected AttachmentService attachmentService;
+    protected WebServiceCredentialService webServiceCredentialService;
+
+    @Override
+    public Attachment createAttachmentFromBackupLink(Document document,
+            AccountingXmlDocumentBackupLink accountingXmlDocumentBackupLink) throws IOException {
+        File formFile = downloadFile(accountingXmlDocumentBackupLink);
+        
+        if (formFile != null) {
+            String uploadFileName = accountingXmlDocumentBackupLink.getFileName();
+            String mimeType = URLConnection.guessContentTypeFromName(formFile.getName());
+            int fileSize = (int) formFile.length();
+            String attachmentType = null;
+            
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("createAttachmentFromBackupLink, uploadFileName: " + uploadFileName + " mimeType: " + mimeType
+                    + " fileSize: " + fileSize);
+            }
+            InputStream fileContents = new FileInputStream(formFile);
+            Attachment attachment = attachmentService.createAttachment(document, uploadFileName, mimeType, fileSize,
+                    fileContents, attachmentType);
+            formFile.delete();
```

Comment:
If you need this temporary file to create the attachment, I would suggest performing the deletion in a "finally" block in case an exception occurs above.

Reply:
ok